lib/class-factory.js

```diff
@@ -139,8 +140,27 @@ function ClassFactory (vm) {
     }
   });
 
+  function getUsedClass(className) {
+    let kclass;
+    while ((kclass = classes[className]) === PENDING_USE) {
+      Thread.sleep(0.05);
+    }
+    if (!kclass) {
+      classes[className] = PENDING_USE;
+    }
+    return kclass;
+  }
+
+  function setUsedClass(className, kclass) {
+    if (kclass) {
+      classes[className] = kclass;
+    } else {
+      delete classes[className];
+    }
+  }
```

Comment:
Let's move these below where they're used. (Code should be readable from top to bottom, higher level details at the top with details further down.)

Reply:
Sorry, I can not find the right place for it. Do you mean only move `delete classes[className]` down, or move the whole function `setUsedClass()` down? 

Function  `setUsedClass()` is used by funtion `use()` and `ensureClass()`, so must be placed outside these two functions. And i also think the code of  `delete classes[className]` should not be move outside `setUsedClass()`.

Pls give me some suggestion.

Reply:
I mean the whole function. Functions can be declared below where they're used. (And ideally as close to the root closure as possible – so they capture as little as possible.)